### PR TITLE
[ALLUXIO-2118] Fix HeartbeatThreadTest

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2118

Previously this test would try to schedule N+1 threads on an executor with N available threads. This would cause DummyHeartbeatTestThread to time out and throw an exception. Unfortunately exceptions from threads aren't propagated, so we never detected this. I've switched to using a Callable for this reason.

I've also lowered the number of iterations so that the test isn't slow.

### CODE REVIEW

File: core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java [hunk 1]
```diff
@@ -76,9 +79,11 @@ public static void afterClass() {
    */
   @Test
   public void serialHeartbeatThreadTest() throws Exception {
-    Thread thread = new DummyHeartbeatTestThread();
+    FutureTask<Null> task = new FutureTask<>(new DummyHeartbeatTestCallable());
```
Comment (jsimsa) [inline core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java]: is there a difference between using `org.apache.commons.lang3.ObjectUtils.Null` and `java.lang.Void`? I have seen `Void` being used in other part of our codebase and I would prefer using that here if it is possible

File: core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java [hunk 2]
```diff
@@ -87,49 +92,48 @@ public void serialHeartbeatThreadTest() throws Exception {
...
-        sExecutorService.submit(ht);
+        mExecutorService.submit(ht);
 
         // Wait for the DummyHeartbeatExecutor executor to be ready to execute its heartbeat.
-        Assert.assertTrue("Initial wait failed.",
+        Assert.assertTrue("Initial wait failed for " + mThreadName,
             HeartbeatScheduler.await(mThreadName, 5, TimeUnit.SECONDS));
 
-        final int numIterations = 100000;
+        final int numIterations = 200;
```
Comment (jsimsa) [inline core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java]: what is the reason for severely reducing the number of iterations here?

**Comment by jsimsa** [comment]
@aaudiber thanks, let some comments

File: core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java (hunk 1 above)
Comment (aaudiber) [inline core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java]: +1 to using Void everywhere

File: core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java (hunk 2 above)
Comment (aaudiber) [inline core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java]: The test takes about 1 second per 10k iterations. I've changed it to 5k iterations now. I don't think this test is worth adding a significant amount of time to the build

Comment (jsimsa) [inline core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java]: ack

**Comment by aaudiber** [comment]
@jsimsa Comments addressed and conflicts resolved

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**